src/simulation/dynamics/constraintEffector/constraintDynamicEffector.h

```diff
@@ -0,0 +1,84 @@
+/*
+ ISC License
+
+ Copyright (c) 2022, Autonomous Vehicle Systems Lab, University of Colorado at Boulder
+
+ Permission to use, copy, modify, and/or distribute this software for any
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+ */
+
+
+#ifndef CONSTRAINT_DYNAMIC_EFFECTOR_H
+#define CONSTRAINT_DYNAMIC_EFFECTOR_H
+
+#include "simulation/dynamics/_GeneralModuleFiles/dynamicEffector.h"
+#include "simulation/dynamics/_GeneralModuleFiles/stateData.h"
+#include "architecture/_GeneralModuleFiles/sys_model.h"
+
+#include "architecture/messaging/messaging.h"
+
+#include "architecture/utilities/bskLogging.h"
+#include "architecture/utilities/avsEigenMRP.h"
+#include <Eigen/Dense>
+#include <vector>
+
+/*! @brief thruster dynamic effector class */
+class ConstraintDynamicEffector: public SysModel, public DynamicEffector {
+public:
+    ConstraintDynamicEffector();
+    ~ConstraintDynamicEffector();
+    void linkInStates(DynParamManager& states);
+    void computeForceTorque(double integTime, double timeStep);
+    void Reset(uint64_t CurrentSimNanos);
+    void UpdateState(uint64_t CurrentSimNanos);
+
+public:   
+    // Gains for PD controller
+    double alpha; //!< [] Baumgarte stabilization gain tuning variable
```

Comment:
This is also important. Let's try to only have ``setters`` and ``getters`` be made public to the user. Everything else should be private.

Reply:
This makes sense, I have changed the module and test/example to use setters/getters instead of making the variables themselves public.

Setters for gains check to make sure those gains are positive, however there are no checks on the vectors that are set since they are allowed to be any magnitude/direction. The setter itself already ensures that only 3 dimensional vectors are passed into the function which is the only restriction required. If these aren't set the default is the zero vector which is also permitted.

I changed the reset function to first check that alpha and beta have been set, then use those to populate individual gains only if an individual gain isn't already set. This way reset no longer has the potential to overwrite previously specified individual gains if reset is called after initialization.